nodl_python/nodl/_parsing/_parsing.py

```diff
@@ -65,3 +64,26 @@ def parse(path: Union[str, Path, IO]) -> List[Node]:
     element_tree = etree.parse(path)
 
     return _parse_element_tree(element_tree)
+
+
+def _parse_multiple(paths: Sequence[Union[str, Path, IO]]) -> List[Node]:
+    """Merge nodl files into one large node list.
+
+    :param paths: List of nodl files to parse
+    :type paths: Sequence[Union[str, Path, IO]]
+    :raises DuplicateNodeError: if node is defined multiple times
+    :raises InvalidNoDLDocumentError: if doc does not adhere to schema
+    :return: flat list of nodes provided by the documents
+    :rtype: List[Node]
+    """
+    node_lists = [parse(path) for path in paths]
+    combined = []
+    combined_dict: Dict[str, List] = defaultdict(list)
+    for node_list in node_lists:
+        for node in node_list:
+            if node.name in combined_dict:
+                if node.executable in combined_dict[node.name]:
```

Comment:
I'm trying to think back to our previous conversation about the executable. Is there a situation where either the node name OR the executable name could be duplicated at all? We're using the AND here, but my initial thought is that if we see either of them twice it will be problematic. Am I missing something?

Reply:
demo_nodes_cpp has two executables which have a 'listener' node in them, 'listener.cpp' and 'listener_best_effort.cpp'. This is a valid use case

Reply:
Ah ha, I knew I was missing something, thank you. Do we actually need to check the node name at all then? Can we just check executable name?

Reply:
fixed in 21f93e1